indico/cli/i18n.py

```diff
@@ -335,6 +338,55 @@ def remove_empty_pot_files(path: Path, python=False, javascript=False, react=Fal
         _get_messages_react_pot(path).unlink()
 
 
+# Merge after generating all the .pot files
+def merge_pot_files(output_file: Path, *input_files: list[Path]):
+    """Merge multiple POT files into a single POT file."""
+    merged_catalog = Catalog(
+        project=DEFAULT_OPTIONS['ExtractMessages']['project'],
+        version=DEFAULT_OPTIONS['ExtractMessages']['version']
+    )
+
+    for input_file in input_files:
+        with input_file.open('rb') as f:
+            catalog = read_po(f)
+
+        for message in catalog:
+            if message.id:
+                merged_catalog[message.id] = message
+
+    with output_file.open('wb') as f:
+        write_po(f, merged_catalog, width=DEFAULT_OPTIONS['ExtractMessages']['width'])
+
+    click.secho('Done merging pot files!', fg='green', bold=True)
+
+
+# Filter merged.po files by removing messages that are not in the given .pot file
+def split_po_by_pot(merged_po_path: Path, pot_path: Path, output_po_path: Path):
+    with merged_po_path.open('rb') as f:
+        merged_catalog = read_po(f)
+
+    with pot_path.open('rb') as f:
+        pot_catalog = read_po(f)
+
+    merged_catalog.update(pot_catalog)
+    merged_catalog.obsolete.clear()
+
+    with output_po_path.open('wb') as f:
+        write_po(f, merged_catalog)
+
+
+def split_all_po_files():
+    for lc_messages_dir in TRANSLATIONS_DIR.glob('*/LC_MESSAGES'):
+        merged_po_path = lc_messages_dir / 'messages-all.po'
+        if merged_po_path.exists():
+            for pot_file in TRANSLATIONS_DIR.glob('*.pot'):
+                if pot_file.name != 'messages-all.pot':
+                    output_po_path = lc_messages_dir / pot_file.name.replace('.pot', '.po')
+                    split_po_by_pot(merged_po_path, pot_file, output_po_path)
```

Comment:
nitpicking, but technically you only need to compute the POT files once before the loop:
```suggestion
    pot_files = [f for f in TRANSLATIONS_DIR.glob('*.pot') if f.name != 'messages-all.pot']
    for lc_messages_dir in TRANSLATIONS_DIR.glob('*/LC_MESSAGES'):
        merged_po_path = lc_messages_dir / 'messages-all.po'
        if merged_po_path.exists():
            for pot_file in pot_files:
                output_po_path = lc_messages_dir / pot_file.name.replace('.pot', '.po')
                split_po_by_pot(merged_po_path, pot_file, output_po_path)
```

Reply:
True! Fixed